source/php-frameworks/symfony.txt

```diff
@@ -0,0 +1,397 @@
+- A terminal app and shell. For MacOS users, use Terminal or a similar app.
+  For Windows users, use PowerShell.
+
+.. _php-symfony-atlas-cluster:
+
+Create a MongoDB Atlas Cluster
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You must create a MongoDB cluster where you can store and manage your
+data. Complete the
+:atlas:`Get Started with Atlas </getting-started>` guide to set up a new
+Atlas account, create a free tier MongoDB cluster, load datasets, and
+interact with the data.
+
+You provide instructions to the driver on where and how to connect to your
+MongoDB cluster by passing a string called the connection string. To retrieve
+your connection string, follow the instructions in the :atlas:`Connect
+to Your Cluster </tutorial/connect-to-your-cluster/>` tutorial in the
+Atlas documentation.
+
+Save your connection string in a secure location.
+
+Initialize a Symfony Project
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Run the following command to create a skeleton Symfony project called
+``restaurants``:
+
+.. code-block:: bash
+
+   composer create-project symfony/skeleton restaurants
+
+Install MongoDB Extension
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Run the following commands to enter your project directory and install
+MongoDB:
+
+.. code-block:: bash
+
+   cd restaurants
+   pecl install mongodb
+
+Install PHP Driver and Doctrine ODM
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Add the MongoDB PHP driver and the MongoDB Doctrine ODM bundle to your
+application. The bundle integrates the ODM library into Symfony so that
+you can read from and write objects to MongoDB. To learn more, see the
+:ref:`php-symfony-resources` section of this guide.
+
+Run the following commands to install the driver and the ODM:
+
+.. code-block:: bash
+
+   composer require mongodb/mongodb
+   composer require doctrine/mongodb-odm-bundle
+
+.. tip::
+
+   After running the preceding commands, you might encounter the
+   following prompt:
+
+   .. code-block:: bash
+      :copyable: false
+
+      Do you want to execute this recipe?
+
+   Select ``yes`` from the response options to add the ODM library to your
+   application.
+
+To ensure that the Doctrine ODM is enabled in your project, verify that your
+``config/bundles.php`` file contains the highlighted entry in the
+following code:
+
+.. code-block:: php
+   :caption: config/bundles.php
+   :emphasize-lines: 3
+   
+   return [
+       // ...
+       Doctrine\Bundle\MongoDBBundle\DoctrineMongoDBBundle::class => ['all' => true],
+   ];
+
+Configure the ODM
+`````````````````
+
+In the ``config/packages`` directory, replace the contents of your
+``doctrine_mongodb.yaml`` file with the following code:
+
+.. code-block:: yaml
+   :caption: config/packages/doctrine_mongodb.yaml
+
+   doctrine_mongodb:
+       auto_generate_proxy_classes: true
+       auto_generate_hydrator_classes: true
+       connections:
+           default:
+               server: '%env(resolve:MONGODB_URL)%&appName=devrel.content.php'
```

Comment:
`&appName=devrel.content.php` doesn't seem relevant to readers at all.

The preceding `&` also assumes that the connection string already has options in it. This may be the case for Atlas, since it tends to add some options to its URLs, but a more robust approach would be to add an extra URI options to the `options` hash on the following line.

Looking at [src/DependencyInjection/DoctrineMongoDBExtension.php](https://github.com/doctrine/DoctrineMongoDBBundle/blob/5.0.1/src/DependencyInjection/DoctrineMongoDBExtension.php#L363) in the bundle, the `options` hash gets used as the second argument to PHPLIB's Client constructor.

Reply:
Removing that option